src/eXpansion/Bundle/Players/Plugins/Gui/ListWindow.php

```diff
@@ -0,0 +1,230 @@
+    public function __construct(
+        $name,
+        $sizeX,
+        $sizeY,
+        $posX = null,
+        $posY = null,
+        WindowFactoryContext $context,
+        PlayerStorage $playerStorage,
+        DataCollectionFactory $dataCollectionFactory,
+        GridBuilderFactory $gridBuilderFactory,
+        ChatCommandDataProvider $chatCommandDataProvider,
+        Connection $connection,
+        AdminGroups $adminGroups,
+        ChatNotification $chatNotification
+
+    ) {
+        parent::__construct($name, $sizeX, $sizeY, $posX, $posY, $context);
+
+        $this->playerStorage = $playerStorage;
+        $this->dataCollectionFactory = $dataCollectionFactory;
+        $this->gridBuilderFactory = $gridBuilderFactory;
+        $this->chatCommandDataProvider = $chatCommandDataProvider;
+        $this->connection = $connection;
+        $this->adminGroups = $adminGroups;
+        $this->chatNotification = $chatNotification;
+    }
+
+    /**
+     * @param mixed $mode
+     */
+    public function setMode($mode)
+    {
+        $this->mode = $mode;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getMode()
+    {
+        return $this->mode;
+    }
+
+    protected function createContent(
+        ManialinkInterface $manialink
+    ) {
+        parent::createContent($manialink);
+        $manialink->setData('mode', $this->getMode());
+        $this->updateData($manialink);
+    }
+
+    protected function updateContent(
+        ManialinkInterface $manialink
+    ) {
+        parent::updateContent($manialink);
+    }
+
+    /**
+     * @param ManialinkInterface $manialink
+     */
+    protected function createGrid(
+        ManialinkInterface $manialink
+    ) {
+
+        $selectButton = $this->uiFactory->createButton('expansion_players.gui.list.window.column.remove',
+            uiButton::TYPE_DEFAULT)->setSize(10, 5)->setTranslate(true);
+
+        $gridBuilder = $this->gridBuilderFactory->create();
+        $gridBuilder->setManialink($manialink)
+            ->setDataCollection($manialink->getData('dataCollection'))
+            ->setManialinkFactory($this)
+            ->addTextColumn(
+                'login',
+                'expansion_players.gui.players.window.column.login',
+                4,
+                true,
+                false
+
+            )
+            ->addActionColumn('login', 'expansion_players.gui.list.window.column.remove',
+                3, [$this, "callbackRemovePlayer"], $selectButton);
+
+
+        $manialink->setData('grid', $gridBuilder);
+
+        $frame = $manialink->getContentFrame();
+        $this->setGridSize($frame->getWidth(), $frame->getHeight() - 10);
+
+        $this->setGridSize(60, 90);
+        $this->setGridPosition(0, 0);
+
+    }
+
+    /**
+     * @param Window|ManialinkInterface $manialink
+     * @throws \Maniaplanet\DedicatedServer\InvalidArgumentException
+     */
+    public function updateData(
+        $manialink
+    ) {
+        $dataset = [];
```

Comment:
Hmm

I would rather have 4 classes extending a single AbstractListWindow and different updateData callback functions instead of having "same" switch case multiple times.